### PR TITLE
feat: author orc id link

### PR DESCRIPTION
**Pull request by codemonkey800**
#525

- Implements the author link component with orc ID icon + link, dashed link styling, and email icon
- Adds author link to dataset description and annotation side panel

## Demos

https://dev-orc-id.cryoet.dev.si.czi.technology/

image

image

### CODE REVIEW

File: frontend/packages/data-portal/app/components/AuthorLegend.tsx [hunk 1]
```diff
@@ -42,7 +42,8 @@ export function AuthorLegend({ inline = false }: { inline?: boolean }) {
     <Tooltip
       title={<Legend />}
       classes={{
-        tooltip: '!p-sds-m !min-w-fit border-solid border border-sds-gray-300',
+        tooltip:
+          '!p-sds-m !min-w-fit border-solid border border-sds-gray-300 !bg-white',
```
[the remark this answers is not in the document]
Comment (tihuan) [reply frontend/packages/data-portal/app/components/AuthorLegend.tsx:46]: Maybe worth adding this as code comment for future reference 👌 ?

File: frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+import { EnvelopeIcon } from 'app/components/icons'
+import { Link } from 'app/components/Link'
+import { cns } from 'app/utils/cns'
+
+import { ORC_ID_URL } from './constants'
+import orcIdImage from './orcid.png'
+import { AuthorInfo } from './types'
+
+const BASE_ICON_SIZE = 10
```
Comment (tihuan) [inline frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx:12]: non blocking nit: Maybe add a suffix of `_PX` for explicitness? 

e.g.,

```ts
const BASE_ICON_SIZE_PX = 10;
```

File: frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                author.primary_author_status
+                  ? 'border-black'
+                  : 'border-sds-gray-500',
+              ]
+            : 'border-transparent',
+        )}
+      >
+        {author.orcid && (
+          <img
+            className="inline"
```
Comment (tihuan) [inline frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx:44]: super nit - the iconseems to be closer to the dotted line compared to the text. Maybe either add some margin-bottom to the icon style or something?

Screenshot 2024-05-13 at 1 44 37 PM

File: frontend/packages/data-portal/app/components/AuthorLink/orcid.png
Comment (tihuan) [inline frontend/packages/data-portal/app/components/AuthorLink/orcid.png:1]: super nit: should we use SVG for smaller image size and better image quality?

File: frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx [hunk 4]
```diff
@@ -1,30 +1,27 @@
...
   return author.name + author.email
 }
 
+const SEPARATOR = `, `
+
 export function DatasetAuthors({
+  AuthorLinkComponent = AuthorLink,
   authors,
   className,
-  separator = ';',
```
Comment (tihuan) [inline frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx:20]: I don't have the context, but I see the separator changed from `;` to `, `, is there backward compatibility / API versioning related things we need to account for?

File: frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx [hunk 5]
```diff
@@ -39,56 +36,56 @@ export function DatasetAuthors({
...
     if (compact && authorsOther.length > 0) {
       if (authorsCorresponding.length === 0) {
         return ellipsis
       }
-      return `${ellipsis} ${separator} `
+      return `${ellipsis} ${SEPARATOR}`
     }
     return null
-  }, [authorsOther, authorsCorresponding, compact, separator])
+  }, [authorsCorresponding.length, authorsOther.length, compact])
```
Comment (tihuan) [inline frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx:49]: nit: I personally don't trust `.length` in the sense that the underlying author objects could change but remain with the same array length. 

Could be an overkill, but maybe a function like `getAuthorIds(authorsCorresponding)` that extract author ids as an array and then run `String()` on it to serialize it, so we get:

```ts
[String(getAuthorIds(authorsCorresponding)), String(getAuthorIds(authorsOther)), compat]
```

File: frontend/packages/data-portal/app/components/Run/AnnotationTable.tsx [hunk 6]
```diff
@@ -185,11 +185,7 @@ export function AnnotationTable() {
             </div>
 
             <div className=" text-sds-gray-600 text-sds-body-xxs leading-sds-header-xxs">
-              <DatasetAuthors
-                authors={annotation.authors}
-                separator=","
```
Comment (tihuan) [inline frontend/packages/data-portal/app/components/Run/AnnotationTable.tsx:190]: nice refactor to extract the separator constant!

**Review by tihuan — APPROVED** [review]
LGTM! Just some non blocking comments 🙆‍♂️ Awesome work, @codemonkey800 !!

File: frontend/packages/data-portal/app/components/AuthorLink/orcid.png
Comment (kne42) [reply frontend/packages/data-portal/app/components/AuthorLink/orcid.png:1]: +1

File: frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  const iconSize = large ? LARGE_ICON_SIZE : BASE_ICON_SIZE
+  const content = (
+    <span className="inline">
+      <span
+        className={cns(
+          'inline border-b',
+
+          author.orcid
+            ? [
+                'border-dashed',
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx:33]: I'll leave this to @Janeece to confirm but I believe it's supposed to switch from dash to solid on hover

File: frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.test.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
+import { render, screen } from '@testing-library/react'
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.test.tsx:1]: yay tests!

**Review by kne42 — COMMENTED** [review]
overall code LGTM, will leave final approval to @Janeece

File: frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.test.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+it('should not be link if orc ID is not provided', () => {
+  render(<AuthorLink author={pick(DEFAULT_AUTHOR, 'name')} />)
+  expect(screen.queryByRole('link')).not.toBeInTheDocument()
+})
+
+it('should be a link if orc ID is provided', () => {
+  render(
+    <AuthorLink
+      author={pick(DEFAULT_AUTHOR, 'name', 'orcid')}
+      LinkComponent={MockLinkComponent}
```
[the remark this answers is not in the document]
Comment (valenzuelaomar) [reply frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.test.tsx:28]: just curious, what issues did you run into locally with `jest. unstable_mockModule`?

File: frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx (hunk 7 above)
Comment (Janeece) [reply frontend/packages/data-portal/app/components/AuthorLink/AuthorLink.tsx:33]: yes, dashed to solid on hover

**Review by Janeece — CHANGES_REQUESTED** [review]
@codemonkey800 this is looking amazing! 3 comments

#### Must-have
- @kne42 already pointed this out, but on hover the underline should go from dashed to solid
- I don't see any corresponding author icons in any dataset or info panels i clicked into, but these should show (sometimes authors have both icons). I'm not sure if this is an issue outside of this PR or not. 

#### Nice to have
- nit - Icon alignment is a bit low on both dataset page and info panels, can we raise them 1-2px so they are optically centered with the letters in the names? Low priority IMO, still looks nice.

File: frontend/packages/data-portal/app/components/AuthorLink/orcid.png
Comment (kne42) [reply frontend/packages/data-portal/app/components/AuthorLink/orcid.png:1]: we already have one here: https://github.com/chanzuckerberg/cryoet-data-portal/blob/main/frontend/packages/data-portal/app/components/icons/ORCIDIcon.tsx

File: frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx (hunk 5 above)
Comment (tihuan) [reply frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx:49]: Awesome! Yeah pretty tricky and annoying lol Also not sure if it'd ever happen given your data, but sometimes I even sort the ids to be SUPER safe 😂

**Review by Janeece — APPROVED** [review]
Looks great, thanks for making those changes @codemonkey800

**Pull request merged**